modules/programs/vdirsyncer.nix

```diff
@@ -0,0 +1,404 @@
+        else let 
+               contents = map (c: if (isString c) 
+                                  then ''"${c}"''
+                                  else mkList (map wrapString c)) cs;
+             in mkList contents; 
+
+      conflictResolutionString = cr: optionalString (cr != null) (
+        if (isString cr) then ''conflict_resolution = "${cr}"''
+                         else ''conflict_resolution = ${mkList (map wrapString (["command"] ++ cr))}''
+      );
+
+      pairString = n: p: ''
+        [pair ${n}]
+        a = "${p.a}"
+        b = "${p.b}"
+        collections = ${collectionsString p.collections}
+        ${conflictResolutionString p.conflictResolution}
+      '';
+
+      formatOption = n: v:
+      if v == null then ""
+      else if (n == "type") then ''type = "${v}"''
+      else if (n == "path") then ''path = "${v}"''
+      else if (n == "fileExt") then ''fileext = "${v}"''
+      else if (n == "encoding") then ''encoding = "${v}"''
+      else if (n == "postHook") then ''post_hook = "${v}"''
+      else if (n == "url") then ''url = "${v}"''
+      else if (n == "timeRange") then ''
+        start_date = "${v.start}"
+        end_date = "${v.end}"
+      ''
+      else if (n == "itemTypes") then ''
+        item_types = ${mkList (map wrapString v)}
+      ''
+      else if (n == "userName") then ''username = "${v}"''
+      else if (n == "password") then ''password = "${v}"''
+      else if (n == "verify") then ''
+        verify = ${if v then "true" else "false"}
+      ''
+      else if (n == "verifyFingerprint") then ''
+        verify_fingerprint = "${v}"
+      ''
+      else if (n == "auth") then ''auth = "${v}"''
+      else if (n == "authCert" && isString(v)) then ''
+        auth_cert = "${v}"
+      ''
+      else if (n == "authCert") then ''
+        auth_cert = ${mkList (map wrapString v)}
+      ''
+      else if (n == "userAgent") then ''useragent = "${v}"''
+      else "";
+
+      storageString = n: s: ''
+        [storage ${n}]
+        ${concatStringsSep "\n" (mapAttrsToList formatOption (filterAttrs (_: v: v != null) s))}
+      '';
+
+    in ''
+       [general]
+       status_path = "${statusPath}"
+
+       ${concatStringsSep "\n" (mapAttrsToList pairString pairs)}
+       ${concatStringsSep "\n" (mapAttrsToList storageString storages)}
+    '');
+in
+
+{
+  options = {
+    programs.vdirsyncer = {
+      enable = mkEnableOption "vdirsyncer";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.vdirsyncer;
+        defaultText = "pkgs.vdirsyncer";
+        description = ''
+          vdirsyncer package to use.
+        '';
+      };
+
+      config = mkOption {
+        default = null;
+        description = ''
+          Configuration for vdirsyncer.
+        '';
+        type = types.nullOr (types.submodule {
+          options = {
+
+            statusPath = mkOption {
+              type = types.str;
+              default = "$HOME/.vdirsyncer/status";
```

Comment:
This defies XDG. I would use XDG_DATA_HOME

Reply:
You’re right, I copied the example from the tutorial, but XDG_DATA_HOME makes more sense.